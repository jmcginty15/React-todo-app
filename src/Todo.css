.Todo, .Todo-edit-form {
    text-align: left;
    margin-bottom: 5px;
}

.Todo-buttons {
    float: right;
}

.Todo-edit, .Todo-remove, .Todo-mark {
    float: none;
}

.Todo-complete {
    text-decoration-line: line-through;
}