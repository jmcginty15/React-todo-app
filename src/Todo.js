import { useState } from 'react';
import './Todo.css';

const Todo = ({ id, task, complete, handleRemove, handleEdit, handleComplete }) => {
    const [isEditing, setIsEditing] = useState(false);
    const [input, setInput] = useState(task);

    function handleRemoveClick() {
        handleRemove(id);
    }

    function showEditForm() {
        setIsEditing(true);
    }

    function handleChange(evt) {
        const newInput = evt.target.value;
        setInput(newInput);
    }

    function handleSubmit(evt) {
        evt.preventDefault();
        if (input) {
            const newTask = input;
            setIsEditing(false);
            setInput(newTask);
            handleEdit(id, newTask);
        }
    }

    function cancelEdit() {
        setIsEditing(false);
        setInput(task);
    }

    function mark() {
        handleComplete(id);
    }

    if (isEditing) return (
        <div className="Todo-edit-form">
            <form onSubmit={handleSubmit}>
                <input name="task" data-testid="edit-todo-input" type="text" placeholder="edit this todo item" value={input} onChange={handleChange} required></input>
                <span className="Todo-buttons">
                    <button type="submit">Submit</button>&nbsp;
                    <button type="button" onClick={cancelEdit}>Cancel</button>
                </span>
            </form>
        </div>
    )

    return (
        <div className="Todo">
            <span data-testid="todo-text" className={complete ? 'Todo-complete' : ''}>{task}</span>
            <span className="Todo-buttons">
                <button className="Todo-mark" onClick={mark}>Mark {complete ? 'Incomplete' : 'Complete'}</button>&nbsp;
                <button className="Todo-edit" onClick={showEditForm}>Edit</button>&nbsp;
                <button className="Todo-remove" onClick={handleRemoveClick}>X</button>
            </span>
        </div>
    )
}

export default Todo;