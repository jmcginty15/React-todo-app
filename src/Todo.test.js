import React from 'react';
import { render } from '@testing-library/react';
import Todo from './Todo';

// smoke test
it('renders without crashing', function () {
    render(<Todo id="1" task="feed dog" complete={false} />);
});

// snapshot test
it('matches snapshot when incomplete', function () {
    const { asFragment } = render(<Todo id="1" task="feed dog" complete={false} />);
    expect(asFragment()).toMatchSnapshot();
});

it('matches snapshot when complete', function () {
    const { asFragment } = render(<Todo id="1" task="feed dog" complete={true} />);
    expect(asFragment()).toMatchSnapshot();
});