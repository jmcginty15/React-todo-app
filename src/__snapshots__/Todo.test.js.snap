// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when complete 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <span
      class="Todo-complete"
      data-testid="todo-text"
    >
      feed dog
    </span>
    <span
      class="Todo-buttons"
    >
      <button
        class="Todo-mark"
      >
        Mark Incomplete
      </button>
       
      <button
        class="Todo-edit"
      >
        Edit
      </button>
       
      <button
        class="Todo-remove"
      >
        X
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when incomplete 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <span
      class=""
      data-testid="todo-text"
    >
      feed dog
    </span>
    <span
      class="Todo-buttons"
    >
      <button
        class="Todo-mark"
      >
        Mark Complete
      </button>
       
      <button
        class="Todo-edit"
      >
        Edit
      </button>
       
      <button
        class="Todo-remove"
      >
        X
      </button>
    </span>
  </div>
</DocumentFragment>
`;
